kernel/posix/pthread_semaphore.c

```diff
@@ -0,0 +1,108 @@
+/*
+ * Copyright (c) 2018 Intel Corporation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <kernel.h>
+#include <pthread.h>
+#include <errno.h>
+
+int sem_destroy(sem_t *semaphore)
+{
+	if (semaphore->sem == NULL) {
+		errno = EINVAL;
+		return -1;
+	}
+
+	k_sem_reset(semaphore->sem);
+
+	return 0;
+}
+
+int sem_getvalue(sem_t *semaphore, int *value)
+{
+	*value = (int) k_sem_count_get(semaphore->sem);
+
+	return 0;
+}
+
+int sem_init(sem_t *semaphore, int pshared, unsigned int value)
+{
+	if (semaphore->sem->limit == 0) {
+		errno = EINVAL;
+		return -1;
+	}
+
+	__ASSERT((pshared < 0), "not correct pshared");
```

Comment:
not correct pshared is not informative, the message needs to say exactly what went wrong. And looking at it, it seems like the assert is wrong, we should be checking for positive numbers and 0, not negative. (this is not an if statement....)